settings/src/main/scala/org/typelevel/sbt/TypelevelSettingsPlugin.scala

```diff
@@ -201,7 +205,56 @@ object TypelevelSettingsPlugin extends AutoPlugin {
         Seq("-Werror")
       else
         Seq.empty
+    },
+    scalacOptions ++= {
+      val (releaseOption, newTargetOption, oldTargetOption) =
+        withJdkRelease(tlJdkRelease.value)(
+          (Seq.empty[String], Seq.empty[String], Seq.empty[String])) { n =>
+          (Seq("-release", n.toString), Seq(s"-target:$n"), Seq("-target:jvm-1.8"))
+        }
+
+      scalaVersion.value match {
+        case V(V(2, 11, _, _)) =>
+          oldTargetOption
+
+        case V(V(2, 12, Some(build), _)) if build >= 5 =>
+          releaseOption ++ oldTargetOption
+
+        case V(V(2, 13, _, _)) =>
+          releaseOption ++ newTargetOption
+
+        case V(V(3, _, _, _)) =>
+          releaseOption
+
+        case _ =>
+          Seq.empty
+      }
+    },
+    javacOptions ++= {
+      withJdkRelease(tlJdkRelease.value)(Seq.empty[String])(n => Seq("--release", n.toString))
     }
   )
 
+  private def withJdkRelease[A](jdkRelease: Option[Int])(default: => A)(f: Int => A): A =
+    jdkRelease.fold(default) {
+      case 8 if isJava8 => default
+      case n if n >= 8 =>
+        if (javaRuntimeVersion < n) {
+          sys.error(
+            s"Target JDK is $n but you are using an older JDK $javaRuntimeVersion. Please switch to JDK >= $n.")
+        } else {
+          f(n)
+        }
+      case n =>
+        sys.error(
+          s"You're using JDK $n, which is not supported by `sbt-typelevel`. Please switch to a newer JDK.")
```

Comment:
`n` is the target JDK right, not the user's runtime JDK?

```suggestion
          s"Target JDK is $n, which is not supported by `sbt-typelevel`. Please select a JDK >= 8.")
```